DrivePID/Drive.cpp

```diff
@@ -0,0 +1,147 @@
+#include "Arduino.h"
+#include "Drive.h"
+
+Drive::Drive(int m1_speed, int m1_p1, int m1_p2,
+             int m2_speed, int m2_p1, int m2_p2,
+             int m3_speed, int m3_p1, int m3_p2) {
+  // parameters for each motor
+  motor_1.set(m1_speed, m1_p1, m1_p2);
+  motor_2.set(m2_speed, m2_p1, m2_p2);
+  motor_3.set(m3_speed, m3_p1, m3_p2);
+}
+
+void Drive::initialize() {
+  motor_1.startMotor();
+  motor_2.startMotor();
+  motor_3.startMotor();
+}
+
+// kinematic equations for robot movement
+void Drive::linealMovement(int degree, int speed) {
+  float m1 = sin(((60 - degree) * PI / 180));
+  float m2 = sin(((180 - degree) * PI / 180));
+  float m3 = sin(((300 - degree) * PI / 180));
+  int speedA = abs(int(m1 * speed));
+  int speedB = abs(int(m2 * speed));
+  int speedC = abs(int(m3 * speed));
```

Comment:
For the part of setting the pwm and direction of the motor, I've already commented in the PR `Drive` that you should encapsulate the functionality in a function inside the `Motor` class, here you could call:
```cpp
motor1.setSpeed(m1*speed);
```

And inside the `Motor` class, something like:

```cpp
void Motor::SetSpeed(int pwm){
    int speed = abs(pwm);
    //set motor speeds
    analogWrite(getMotorSpeed(), speed);
    if (pwm >= 0) {
        motorFrontward();
    } else {
        motorBackward();
}
```